### PR TITLE
[MPS] Reject embedding_bag index equal to num_weights

### LINKED ISSUE #189971: [MPS] embedding_bag accepts index == num_embeddings and reads one row past the weight table
### 🐛 Describe the bug

The MPS `embedding_bag` weight-index bounds check is off by one: the kernel
uses `> num_weights` instead of `>= num_weights`, so index exactly equal to
`num_embeddings` silently reads one row past the weight table. Backward can
apply the same invalid row as an out-of-range gradient write.

```python
import torch

# Weight table = first 5 rows of a 6-row buffer, so the out-of-bounds row
# (index 5 == num_weights) lands on known adjacent data big[5] = [15,16,17].
big = torch.arange(18.0).reshape(6, 3)
W = big.to("mps")[:5]                     # num_weights = 5, valid idx 0..4
idx = torch.tensor([5], device="mps")     # == num_weights
offs = torch.tensor([0], device="mps")
out = torch.nn.functional.embedding_bag(idx, W, offs, mode="sum")
torch.mps.synchronize()
print("MPS returned:", out.cpu().tolist())  # [[15.0, 16.0, 17.0]] -- the row past the table
```

Index 6 correctly raises (`Index 0 is out of bounds: 6, range 0 to 5`),
pinning the failure to exactly the equality boundary. CPU raises for index 5.
Output on a source build of current main (2.14.0a0+git2662808); also
reproduces on 2.13.0 stable.

The guard is at
https://github.com/pytorch/pytorch/blob/26628087023e5102381849e9a27c3f09b6d485a4/aten/src/ATen/native/mps/kernels/EmbeddingBag.metal#L185-L194
-- `static_cast<uint32_t>(weight_idx) > num_weights` passes `weight_idx ==
num_weights`, and the subsequent row read uses it unchecked.

History: #163630 reported missing index validation and #168930 added this
check, but with `>` rather than `>=`; its regression test used index 6
against a 4-row table, which does not cover the equality boundary.

### Versions

PyTorch 2.14.0a0+git2662808 (source build of main) and 2.13.0; macOS 26.5.2 (arm64), Apple M5 Max, Python 3.12.13.

cc @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk @Isalia20

### PR DESCRIPTION
**Pull request by CodersAcademy006**
Fixes #189971.

## Root cause
The MPS `embedding_bag` forward bounds check in `aten/src/ATen/native/mps/kernels/EmbeddingBag.metal` used `static_cast<uint32_t>(weight_idx) > num_weights`. Valid weight indices are `0 .. num_weights - 1`, so an index exactly equal to `num_weights` passed the check and the kernel read one row past the weight table. The backward path applies the same index as an out-of-range gradient write.

## Fix
Change the comparison to `>= num_weights` so the exact upper boundary is rejected like any other out-of-range index. This matches CPU and CUDA, which reject `index == num_weights`. The forward gate is sufficient because the backward accumulation only runs for indices the forward already validated.

## Test
Extended `test_embedding_bag_out_of_bounds` in `test/test_mps.py` with a boundary case where the index equals `num_weights`; it now raises instead of silently reading out of bounds.

```
python test/test_mps.py -k test_embedding_bag_out_of_bounds
```

### CODE REVIEW

**Comment by CodersAcademy006** [comment]
Closing as a duplicate: #189973 by @bogatyy already fixes this same off-by-one bounds check and predates this PR. Deferring to it. Apologies for the duplicate.
